sale_order_weight/models/sale_order.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2021 Manuel Calero Solís (https://xtendoo.es)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    total_weight = fields.Float(
+        compute='_compute_total_weight',
+        string='Total Weight',
+        store=True
+    )
+
+    @api.depends('order_line.weight')
+    def _compute_total_weight(self):
+        for order in self:
+            total_weight = 0.0
+            for line in order.order_line:
+                total_weight += line.weight
+            order.total_weight = total_weight
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    weight = fields.Float(compute='_compute_weight', string='Weight', store=True)
```

Comment:
Hi. This field should not be a computed field, but a field populated by an ochange in my opinion. 
If you change the weight of a product, it should not change the weight of the older orders. (same mechanism as for the price).

don't you think ?